src/testing/flow_solver_cases/periodic_1D_flow.cpp

```diff
@@ -0,0 +1,125 @@
+#include "periodic_1D_flow.h"
+#include <deal.II/base/function.h>
+#include <stdlib.h>
+#include <iostream>
+#include <deal.II/dofs/dof_tools.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/numerics/vector_tools.h>
+#include <deal.II/fe/fe_values.h>
+#include "physics/physics_factory.h"
+#include "dg/dg.h"
+#include "mesh/grids/straight_periodic_cube.hpp"
+#include <deal.II/base/table_handler.h>
+//#include <cmath>
+
+namespace PHiLiP {
+
+namespace Tests {
+//=========================================================
+// PERIODIC 1D DOMAIN FOR TIME REFINEMENT STUDY
+//=========================================================
+template <int dim, int nstate>
+Periodic1DFlow<dim, nstate>::Periodic1DFlow(const PHiLiP::Parameters::AllParameters *const parameters_input)
+        : FlowSolverCaseBase<dim, nstate>(parameters_input)
+        , number_of_refinements(this->all_param.grid_refinement_study_param.num_refinements)
+        , domain_left(this->all_param.grid_refinement_study_param.grid_left)
+        , domain_right(this->all_param.grid_refinement_study_param.grid_right)
+        , unsteady_data_table_filename_with_extension(parameters_input->flow_solver_param.unsteady_data_table_filename+".txt")
+{
+    // Get the flow case type -- I think this was for assigning the TGV stuff
+    //using FlowCaseEnum = Parameters::FlowSolverParam::FlowCaseType;
+    //const FlowCaseEnum flow_type = parameters_input->flow_solver_param.flow_case_type;
+
+    //this->number_times_refined_by_half = 0;
+}
+
+template <int dim, int nstate>
+void Periodic1DFlow<dim,nstate>::display_flow_solver_setup(std::shared_ptr<InitialConditionFunction<dim,nstate,double>> /*initial_condition*/) const
+{
+    using PDE_enum = Parameters::AllParameters::PartialDifferentialEquation;
+    const PDE_enum pde_type = this->all_param.pde_type;
+    std::string pde_string;
+    if (pde_type == PDE_enum::advection)                {pde_string = "advection";}
+    this->pcout << "- PDE Type: " << pde_string << std::endl;
+    this->pcout << "- Polynomial degree: " << this->all_param.grid_refinement_study_param.poly_degree << std::endl;
+    this->pcout << "- Constant time step size: " << this->all_param.ode_solver_param.initial_time_step << std::endl;
+    this->pcout << "- Final time: " << this->all_param.flow_solver_param.final_time << std::endl;
+
+}
+
+template <int dim, int nstate>
+std::shared_ptr<Triangulation> Periodic1DFlow<dim,nstate>::generate_grid() const
+{
+    std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation> (
+#if PHILIP_DIM!=1
+            this->mpi_communicator
+#endif
+    );
+    
+    dealii::GridGenerator::hyper_cube(*grid, domain_left, domain_right, true); //colorize = true
```

Comment:
This could be moved to `src/mesh/grids/straight_periodic_cube.cpp`

Reply:
Consider modifying the existing `periodic_cube_flow.cpp` and using the `is_taylor_green_vortex` flag to turn off things that do now pertain to your flow case because there is a lot of repeated code

Reply:
I will move things into `periodic_cube_flow.cpp` (or renamed) after I've addressed other comments.

Mesh has been moved to `straight_periodic_cube.cpp`. I also added 2D square domain to make it complete.

Reply:
Ok I'll resolve it once its all complete

Reply:
Btw I think we should just leave it named as `PeriodicCubeFlow` since we're enforcing all domain sides to be of equal length